### PR TITLE
HDDS-4176. Fix failed UT: test2WayCommitForTimeoutException

### PR DESCRIPTION
**Pull request by runzhiwang**
## What changes were proposed in this pull request?
**What's the problem ?**
image

**What's the reason of the failed ut ?**

When cluster create 9 nodes, the following code try to shutdown the follower of the pipeline. Actually, the pipeline only exist in 3 nodes, but the following code check 9 nodes, so the pipeline can not found in the other 6 nodes. 
```
    for (HddsDatanodeService dn : cluster.getHddsDatanodes()) {
      // shutdown the ratis follower
      if (ContainerTestHelper.isRatisFollower(dn, pipeline)) {
        cluster.shutdownHddsDatanode(dn.getDatanodeDetails());
        break;
      }
    }
```

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4176

## How was this patch tested?

Existed ut.

### CODE REVIEW

**Comment by runzhiwang** [comment]
@elek Could you help review this patch ? Thank you very much.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestWatchForCommit.java [hunk 1]
```diff
@@ -297,9 +298,11 @@ public void test2WayCommitForTimeoutException() throws Exception {
             xceiverClient.getPipeline()));
     reply.getResponse().get();
     Assert.assertEquals(3, ratisClient.getCommitInfoMap().size());
+    List<DatanodeDetails> datanodeDetails = pipeline.getNodes();
     for (HddsDatanodeService dn : cluster.getHddsDatanodes()) {
       // shutdown the ratis follower
-      if (ContainerTestHelper.isRatisFollower(dn, pipeline)) {
+      if (datanodeDetails.contains(dn.getDatanodeDetails())
+          && ContainerTestHelper.isRatisFollower(dn, pipeline)) {
```
Comment (amaliujia) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestWatchForCommit.java:305]: This logic looks right to me.

Can you bump up the number of DN to make tests run against higher number DN: https://github.com/apache/hadoop-ozone/blob/34ee8311b0d0a37878fe1fd2e5d8c1b91aa8cc8f/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/Test2WayCommitInRatis.java#L109

Which seems will cover your change?

Comment (runzhiwang) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestWatchForCommit.java:305]: @amaliujia Thanks for your review. TestWatchForCommit already startCluster with 9 nodes, maybe it is enough.
https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestWatchForCommit.java#L124

Comment (amaliujia) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestWatchForCommit.java:305]: Oops. You are right. I was looking at a wrong file.

**Review by adoroszlai — APPROVED** [review]
Thanks @runzhiwang for fixing this long standing intermittent failure.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestWatchForCommit.java [hunk 2]
```diff
@@ -297,9 +298,11 @@ public void test2WayCommitForTimeoutException() throws Exception {
             xceiverClient.getPipeline()));
     reply.getResponse().get();
     Assert.assertEquals(3, ratisClient.getCommitInfoMap().size());
+    List<DatanodeDetails> datanodeDetails = pipeline.getNodes();
```
[the remark this answers is not in the document]
Comment (runzhiwang) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestWatchForCommit.java]: @adoroszlai Thanks for review. I have updated the patch.

**Comment by adoroszlai** [comment]
Thanks @runzhiwang for the fix and @amaliujia for the review.

**Pull request merged**